modules/standard/Path.chpl

```diff
@@ -810,6 +813,85 @@ proc file.relPath(start:string=curDir): string throws {
   return Path.relPath(this.path, start);
 }
 
+/*
+  Returns a new path with dirname replaced with the provided
+  new argument of dirname. If path had no dirname the argument is
+  added to the path.
+
+  :arg path: `string` Original Path.
+  :newDirname path: `string` for the new dirname
+
+  :returns: The new path after replacing dirname.
+  :rtype: `string`
+
+*/
+proc replaceDirname(path: string, newDirname: string): string {
+    const (dirname, basename) = splitPath(path);
+    return joinPath(newDirname, basename);
+}
+
+/*
+  Returns a new path with basename replaced with the provided
+  new argument of basename. If path had no basename the argument is
+  added to the path or if the provided new basename is empty string then basename is removed from path.
```

Comment:
Could you make sure that your lines fit in 80 columns?

Reply:
corrected them to maintain 80 columns limit